### ISSUE TITLE
ability to specify different exception instead of custom `SocketBlockedError`

### ISSUE THREAD

**Issue by fopinappb** [miketheman/pytest-socket#67]
Some packages and their dependencies (such as `tldextract` using `requests`) are prepared to handle connection issues with standard exceptions.

Raising a custom one forces us to mock things that are not really important, if it'd be ok to let them fail.

Having the ability somehow to choose a different exception to be raised, such as standard/builtin `ConnectionRefusedError` would be nice.

I'm currently monkey patching pytest-socket in my `conftest.py` for this:

```
def pytest_runtest_setup():
    # monkey patch pytest_socket exception for a friendlier one
    # as some libs will react badly to RuntimeErrors
    import pytest_socket

    pytest_socket.SocketBlockedError = ConnectionRefusedError
```

**Comment by miketheman** [comment]
Hey @fopinappb ! Thanks for reporting - this is an interesting use case. Do you think that having `SocketBlockedError` inherit from `ConnectionRefusedError` instead of `RuntimeError` would work here?

The original choice of RuntimeError was to raise the issue to the visibility of the operator that they had expressed a test condition without explicitly allowing it.

I was looking for `ConnectionRefusedError` in tldextract and couldn't find it - could you share where you're seeing that exception raised?

**Comment by fopinappb** [comment]
hey @miketheman I've been monkeypatching socket before I found this package (which I prefer now, for main tenability).
I also used a custom RuntimeError at first to easily spot the "unexpected internet usage" but then there were all these packages that used requests or urllib3 in background and handled `requests` exceptions but not the rest.

And `RuntimeError` are not supposed to be handled so requests/urllib3 do not indeed..

Switching to `ConnectionRefusedError` (as an official exception thrown by `socket.socket`) did sort it out, but I haven't checked exactly where...

For `tldextract` specifically I believe it's here https://github.com/john-kurkowski/tldextract/blob/0b876fa863bc54fa86c2a4d8b4f58391873e2281/tldextract/suffix_list.py#L33-L34

A RuntimeError won't be a requests exception and won't land there.

Custom subclass of `ConnectionRefusedError` should work. Can't be sure, but I guess it will land here in requests https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/adapters.py#L497-L498

so any socket error will be handled gracefully
